xbmc/guilib/StereoscopicsManager.cpp

```diff
@@ -99,29 +99,38 @@ static const struct StereoModeMap StringToGuiModeMap[] =
 };
 
 
-CStereoscopicsManager::CStereoscopicsManager(void)
+CStereoscopicsManager::CStereoscopicsManager(CSettings &settings,
+                                             CDataCacheCore &dataCacheCore,
+                                             CGUIWindowManager &windowManager) :
```

Comment:
this is not good either. don't try to introduce DI through the back door

Reply:
How should this be done?

Reply:
like it is done everywhere else. use CServiceBroker::GetDataCache() at those location where you need it. 
for CGUIWindowManager this creates a hard dependency to this ojects. actually the member is only used once for windowManager.AddMsgTarget(this)
AddMsgTarget is someting we need to change. For now you can register StereoScopicManager after you create it in GUIComponent.